library/hls/src/test/java/com/google/android/exoplayer2/source/hls/HlsMediaChunkTest.java

```diff
@@ -0,0 +1,260 @@
+    HlsMediaPlaylist hlsMediaPlaylist = HlsTestUtils.getHlsMediaPlaylist(PLAYLIST_INDEPENDENT_PART, variant2);
+
+    // First Part checks segment level independent
+    HlsChunkSource.SegmentBaseHolder nextSegmentBaseHolder =
+        new HlsChunkSource.SegmentBaseHolder(hlsMediaPlaylist.trailingParts.get(0), 0, 0);
+    boolean result =
+        HlsMediaChunk.shouldSpliceIn(previousChunk, variant2, hlsMediaPlaylist, BASE_VIDEO_FORMAT, nextSegmentBaseHolder, 0);
+
+    assertThat(result).isFalse();
+
+    // Additional Parts must be themselves independent, regardless of the independence of the playlist
+    nextSegmentBaseHolder =
+        new HlsChunkSource.SegmentBaseHolder(hlsMediaPlaylist.trailingParts.get(1), 1, 1);
+    result =
+        HlsMediaChunk.shouldSpliceIn(previousChunk, variant2, hlsMediaPlaylist, BASE_VIDEO_FORMAT, nextSegmentBaseHolder, 0);
+    assertThat(result).isFalse();
+
+  }
+
+  @Test
+  public void test_shouldSpliceIn_IntraTrickPlay() {
+
+    // Trick play to trick play track should never need a splice, even overlapping
+
+    Uri variant4 = Uri.parse("http://example.com/iframe4.m3u8");
+    HlsMediaPlaylist hlsMediaPlaylist4s = HlsTestUtils.getHlsMediaPlaylist(PLAYLIST_IFRAME_4s, variant4);
+    HlsChunkSource.SegmentBaseHolder previousSegmentBaseHolder =
+        new HlsChunkSource.SegmentBaseHolder(hlsMediaPlaylist4s.segments.get(0), 0, 0);
+    HlsMediaChunk previousChunk = createTestHlsMediaChunk(IFRAME_FORMAT, previousSegmentBaseHolder, variant4, true);
+
+    Uri variant2 = Uri.parse("http://example.com/iframe2.m3u8");
+    HlsMediaPlaylist hlsMediaPlaylist2s = HlsTestUtils.getHlsMediaPlaylist(PLAYLIST_IFRAME_2s, variant2);
+    HlsChunkSource.SegmentBaseHolder nextSegmentBaseHolder =
+        new HlsChunkSource.SegmentBaseHolder(hlsMediaPlaylist2s.segments.get(1), 0, 0);
+    boolean result =
+        HlsMediaChunk.shouldSpliceIn(previousChunk, variant2, hlsMediaPlaylist2s, IFRAME_FORMAT, nextSegmentBaseHolder, 0);
+
+    assertThat(result).isFalse();
+  }
+
+  @Test
+  @Ignore
+  public void test_shouldSpliceIn_NonTrickPlay_To_TrickPlay() {
+
+    // Switch to a trick-play track never requires splice in, even if overlapping
+    HlsChunkSource.SegmentBaseHolder previousSegmentBaseHolder =
+        new HlsChunkSource.SegmentBaseHolder(playlist.segments.get(0), 0, 0);
+    HlsMediaChunk previousChunk = createTestHlsMediaChunk(BASE_VIDEO_FORMAT, previousSegmentBaseHolder, PLAYLIST_URI, true);
+
+    Uri variant2 = Uri.parse("http://example.com/iframe2.m3u8");
+    HlsMediaPlaylist hlsMediaPlaylist2s = HlsTestUtils.getHlsMediaPlaylist(PLAYLIST_IFRAME_2s, variant2);
+    HlsChunkSource.SegmentBaseHolder nextSegmentBaseHolder =
+        new HlsChunkSource.SegmentBaseHolder(hlsMediaPlaylist2s.segments.get(1), 0, 0);
+    boolean result =
+        HlsMediaChunk.shouldSpliceIn(previousChunk, variant2, hlsMediaPlaylist2s, IFRAME_FORMAT, nextSegmentBaseHolder, 0);
+
+    assertThat(result).isFalse();
```

Comment:
> So the splice will effectively discard the downloaded "lower res" media like evaluateQueueSize() with discard does in favor of the adaptive switch occurs? 

If this is true (more dense samples from non-trickplay overwrite (replace) the less dense trick-play samples then yes, I agree I need to add a test case and support trick-play to non-trick play mode.  We are handling this non-adaptively now by forcing a seek and a completely new `TrackGroup` (so sample queue is flushed), but it would be better if this could happen entirely in the adaptive switch.  For obvious reasons you would not want to continue with a very low frame rate set of samples after switching back to normal playback.